src/credentials/tests/TestGroupDataProvider.cpp

```diff
@@ -132,6 +132,19 @@ class TestListener : public GroupDataProvider::GroupListener
 };
 static TestListener sListener;
 
+bool CompareKeySets(const KeySet & keyset1, const KeySet & keyset2)
+{
+    VerifyOrReturnError(keyset1.policy == keyset2.policy, false);
+    VerifyOrReturnError(keyset1.num_keys_used == keyset2.num_keys_used, false);
+    VerifyOrReturnError(keyset1.epoch_keys[0].start_time == keyset2.epoch_keys[0].start_time, false);
+    VerifyOrReturnError(keyset1.epoch_keys[1].start_time == keyset2.epoch_keys[1].start_time, false);
+    VerifyOrReturnError(keyset1.epoch_keys[2].start_time == keyset2.epoch_keys[2].start_time, false);
```

Comment:
We store derived operational group keys. Should `epoch_keys` not be renamed to `keys`, since epoch keys now only appear at ingestion, but at usage time, it's operational group keys?

Reply:
The operational keys are internal. Since this is a public API, what the caller write and read are `epoch_keys`. The returned epoch_key values are always NULL, but the start_time can still be read back.